electrum/lnworker.py

```diff
@@ -1925,7 +1925,7 @@ def suggest_splits(
         r_tags,
     ) -> List['SplitConfigRating']:
         channels_with_funds = {
-            (chan.channel_id, chan.node_id): int(chan.available_to_spend(HTLCOwner.LOCAL))
+            (chan.channel_id, chan.node_id): int(chan.available_to_spend(HTLCOwner.LOCAL, single_payment=True))
```

Comment:
I appreciate the intent and this is going in the right direction but note that suggest_splits might decide to add multiple htlcs onto a given channel, not just one, and this test only checks whether a single htlc can be added. That is, if it decides to split the payment amount within a single channel, the check is not sufficient.

Reply:
@accumulator I believe @SomberNight's comment was about fixing #8857, by checking `max_accepted_htlcs`

Reply:
Ah yes, you can read that in multiple ways :)

Anyway, that would be for `suggest_splits` to check I think. 

My initial PR had `determine_payment_constraints` for that reason, to have available the remaining # of htlcs for a channel, and the minimum msat amount per htlc, not just max amount.